File: 1002/1002.lua
convert_table = {"ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu"}
result = ""

function number2pinyin(naturalNum, i)

	-- sum
	tmpNum = 0;
	list = num2list(naturalNum)
	for i=1, #list do
		tmpNum = tmpNum + list[i]
	end

	-- convert
	list = num2list(tostring(tmpNum))
	for i=1, #list do
		result = result .. convert_table[list[i]+1] .. " "
	end

	return string.sub(result, 1, string.len(result) - 1)
end

function num2list(str)
	list = {}
	for i=1, string.len(str) do
		list[i] = tonumber(string.sub(str, i, i))
	end
	return list
end

-- print("please enter a integer: ")
string = tostring(io.read("*l"))
print(number2pinyin(string))

File: 1002/1002.py
# coding:utf-8
convert_table = ("ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu")

def sum2pinyin(string):
	# sum
	tmpNum = 0
	for x in xrange(0,len(string)):
		tmpNum = tmpNum + int(string[x])

	# convert
	result = ""
	tmpNum = str(tmpNum)
	for x in xrange(0,len(tmpNum)):
		result = result + convert_table[int(tmpNum[x])] + " "

	return result[:len(result) - 1]

# print "please enter a integer: "
string = raw_input()
print sum2pinyin(string)